packages/amplication-client/src/User/SignupPreviewAccount.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect } from "react";
+import { useHistory } from "react-router-dom";
+import { gql, useMutation } from "@apollo/client";
+import { formatError } from "../util/error";
+import "./Signup.scss";
+import { setToken } from "../authentication/authentication";
+
+const SignupPreviewAccount = () => {
+  const history = useHistory();
+
+  const [signupPreviewAccount, { loading, error }] = useMutation<{
+    signupPreviewAccount: {
+      token: string;
+      workspaceId: string;
+      resourceId: string;
+      projectId: string;
+    };
+  }>(SIGNUP_PREVIEW_ACCOUNT);
+
+  useEffect(() => {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const email = urlParams.get("email");
+    const previewAccountType = urlParams.get("previewAccountType");
+
+    if (email && previewAccountType) {
+      signupPreviewAccount({
```

Comment:
Move this logic to a separated hook and expose a function that received the relevant params